site/cds_rdm/tasks.py

```diff
@@ -140,3 +147,224 @@ def sync_users():
     except Exception as e:
         db.session.rollback()
         current_app.logger.exception(e)
+
+
+@shared_task()
+def sync_local_accounts_to_names(since, user_id=None):
+    """Syncs local accounts to names vocabulary.
+
+    If the name is marked as "unlisted" - meaning it is deprecated - we still update it with the new values, if needed.
```

Comment:
Right now we only have one use case where an user is considered deprecated. This happens when we have the user is coming from ORCID and coming from CERN database with the ORCID registered there as well, in these cases we mark the CERN user as deprecated and merge the extra data in the ORCID value.

However, there can be more cases in the future where we want to deprecate names, not sure about the one you mention - when a user leaves CERN - is a valid one, as per my understanding we might want to still add that person as contributor of a project although they have left. However it might be a valid case or there might be others.

Existing records are not affected by deprecated values, this only affects the search display,  the depreacted values won't be shown.